python/tank/bootstrap/configuration_writer.py

```diff
@@ -0,0 +1,474 @@
+            # bake in which version of the deploy logic was used to push this config
+            metadata["host"] = self._sg_connection.base_url
+            # and include details about where the config came from
+            metadata["http_proxy"] = self._sg_connection.config.raw_http_proxy
+            # write yaml
+            yaml.safe_dump(metadata, fh)
+            fh.write("\n")
+            fh.write("# End of file.\n")
+
+        log.debug("Wrote %s" % sg_file)
+
+    def write_pipeline_config_file(self, pipeline_config_id, project_id, entry_point, bundle_cache_fallback_paths):
+        """
+        Writes out the config/core/pipeline_config.yml
+        """
+        # the pipeline config metadata
+        # resolve project name and pipeline config name from shotgun.
+        if self._pipeline_config_id:
+            # look up pipeline config name and project name via the pc
+            log.debug("Checking pipeline config in Shotgun...")
+
+            sg_data = self._sg_connection.find_one(
+                constants.PIPELINE_CONFIGURATION_ENTITY_TYPE,
+                [["id", "is", pipeline_config_id]],
+                ["code", "project.Project.tank_name"]
+            )
+
+            project_name = sg_data["project.Project.tank_name"] or constants.UNNAMED_PROJECT_NAME
+            pipeline_config_name = sg_data["code"] or constants.UNMANAGED_PIPELINE_CONFIG_NAME
+
+        elif project_id:
+            # no pc. look up the project name via the project id
+            log.debug("Checking project in Shotgun...")
+
+            sg_data = self._sg_connection.find_one(
+                "Project",
+                [["id", "is", project_id]],
+                ["tank_name"]
+            )
+
+            project_name = sg_data["tank_name"] or constants.UNNAMED_PROJECT_NAME
+            pipeline_config_name = constants.UNMANAGED_PIPELINE_CONFIG_NAME
+
+        else:
+            project_name = "Site"
+            pipeline_config_name = constants.UNMANAGED_PIPELINE_CONFIG_NAME
+
+        pipeline_config_content = {
+            "pc_id": pipeline_config_id,
+            "pc_name": pipeline_config_name,
+            "project_id": project_id,
+            "project_name": project_name,
+            "entry_point": entry_point,
+            "published_file_entity_type": "PublishedFile",
+            "use_bundle_cache": True,
+            "bundle_cache_fallback_roots": bundle_cache_fallback_paths,
+            "use_shotgun_path_cache": True
+        }
+
+        # write pipeline_configuration.yml
+        pipeline_config_path = os.path.join(
+            self._path.current_os,
+            "config",
+            "core",
+            constants.PIPELINECONFIG_FILE
+        )
+
+        with self._open_auto_created_yml(pipeline_config_path) as fh:
+            yaml.safe_dump(pipeline_config_content, fh)
+            fh.write("\n")
+            fh.write("# End of file.\n")
+
+    def write_bare_pipeline_config_file(self, entry_point):
```

Comment:
The contents of this method are pretty similar to the end of the previous method. Maybe there's one method to write a config and these two methods just pass in the content dict?